plugins/camera/camera.h

```diff
@@ -0,0 +1,381 @@
+     * @return Result of request.
+     */
+    Result start_photo_interval(float interval_s);
+
+    /**
+     * @brief Stop photo interval (synchronous).
+     *
+     * Stops a photo timelapse again.
+     *
+     * @return Result of request.
+     */
+    Result stop_photo_interval();
+
+    /**
+     * @brief Start video capture (synchronous).
+     *
+     * This starts a video recording immediately.
+     *
+     * @return Result of request.
+     */
+    Result start_video();
+
+    /**
+     * @brief Stop video capture (synchronous).
+     *
+     * This stops a video recording again.
+     *
+     * @return Result of request.
+     */
+    Result stop_video();
+
+    /**
+     * @brief Take photo (asynchronous).
+     *
+     * This takes one photo immediately.
+     *
+     * @param callback Function to call with result of request.
+     */
+    void take_photo_async(const result_callback_t &callback);
+
+    /**
+     * @brief Start photo interval (asynchronous).
+     *
+     * Starts a photo timelapse given an interval.
+     *
+     * @param interval_s The interval between photos in seconds.
+     * @param callback Function to call with result of request.
+     */
+    void start_photo_interval_async(float interval_s, const result_callback_t &callback);
+
+    /**
+     * @brief Stop photo interval (asynchronous).
+     *
+     * Stops a photo timelapse again.
+     *
+     * @param callback Function to call with result of request.
+     */
+    void stop_photo_interval_async(const result_callback_t &callback);
+
+    /**
+     * @brief Start video capture (asynchronous).
+     *
+     * This starts a video recording immediately.
+     *
+     * @param callback Function to call with result of request.
+     */
+    void start_video_async(const result_callback_t &callback);
+
+    /**
+     * @brief Stop video capture (asynchronous).
+     *
+     * This stops a video recording again.
+     *
+     * @param callback Function to call with result of request.
+     */
+    void stop_video_async(const result_callback_t &callback);
+
+
+    /**
+     * @brief Camera mode type.
+     */
+    enum class Mode {
+        PHOTO,
+        VIDEO,
+        UNKNOWN
+    };
+
+    /**
+     * @brief Callback type for asynchronous camera mode calls.
+     */
+    typedef std::function<void(Result, const Mode &)> mode_callback_t;
+
+    /**
+     * @brief Setter for camera mode (asynchronous).
+     *
+     * @param mode Camera mode to set.
+     * @param callback Function to call with result of request.
+     */
+    void set_mode_async(Mode mode, const mode_callback_t &callback);
+
+    /**
+     * @brief Getter for camera mode (asynchronous).
+     *
+     * @param callback Function to call with result of request.
+     */
+    void get_mode_async(const mode_callback_t &callback);
+
+    /**
+     * @brief Information about a picture just captured.
+     */
+    struct CaptureInfo {
+        /**
+         * @brief Position type.
```

Comment:
Didn't position and Quaternion recently get pulled out of telemetry class? If so, are they reusable?

Reply:
It got pulled out of the telemetry class but not out of the plugin. And we don't want dependencies between plugins, so for now this is duplicated. If we decide to pull such types out of the plugins into something common, then we can indeed re-use it.
It's always a trade-off between dependencies and duplication.